src/modules/navigator/mission_block.cpp

```diff
@@ -251,15 +251,26 @@ MissionBlock::is_mission_item_reached()
 				}
 
 			} else {
+				// Determine altitude acceptance radius. By default, this is taken from the respective parameter.
+				// However, before a landing approach the acceptance radius needs to be tighter: Assume a 30% error
+				// w.r.t. the remaining descent altitude is OK, but enforce min/max values (e.g. min=3m to make
+				// sure that the waypoint can still be reached in case of wrong user input).
+				float alt_accept_rad = _navigator->get_altitude_acceptance_radius();
```

Comment:
Looking at how navigator overall works it would be cleaner to move this into a new `MissionBlock::get_altitude_acceptance()` that checks the vehicle type and setpoint triplet. All callers except for one (feasibility) are MissionBlocks.

`Navigator::get_altitude_acceptance_radius()` could become `Navigator::get_default_altitude_acceptance_radius()` for grabbing the param values.

That would also give us a good mechanism for tightening the altitude acceptance during takeoff.

Reply:
Changed, but a bit differently than you proposed (i thought it is simpler that way because we don't need new functions). Does this "different" variant still work for you?